test/core/tsi/s2a/s2a_test_util.cc

```diff
@@ -185,3 +187,220 @@ size_t expected_message_size(size_t plaintext_size) {
   return 5u /* header */ + plaintext_size + 16u /* tag */ +
          1u /* record type */;
 }
+
+bool check_encrypt_record(TLSCiphersuite ciphersuite, uint8_t* record_one,
+                          size_t record_one_size, uint8_t* record_two,
+                          size_t record_two_size, uint8_t* record_three,
+                          size_t record_three_size, char** error_details) {
+  if (record_one == nullptr && record_one_size != 0) {
+    *error_details = gpr_strdup(
+        "If |record_one| is nullptr, then |record_one_size| must be zero.");
+    return false;
+  }
+  if (record_two == nullptr && record_two_size != 0) {
+    *error_details = gpr_strdup(
+        "If |record_two| is nullptr, then |record_two_size| must be zero.");
+    return false;
+  }
+  if (record_three == nullptr && record_three_size != 0) {
+    *error_details = gpr_strdup(
+        "If |record_three| is nullptr, then |record_three_size| must be zero.");
+    return false;
+  }
+  switch (ciphersuite) {
```

Comment:
These three `switch` statements seem to have very similar logic. Can we have that logic in a separate function that we can three times here?

Reply:
I turned the body of the cases into a standalone function, `verify_record`, that is called in each case with different parameters.